server/spec.go

```diff
@@ -94,6 +100,9 @@ type ResponseError interface {
 // ResponseWriter is a wrapper for http.ResponseWriter to track the written
 // status code.
 type ResponseWriter interface {
+	// BodyBuffer returns the buffer which is used to track the bytes being
+	// written to the response.
+	BodyBuffer() *bytes.Buffer
```

Comment:
This extension is necessary for the transaction responder. We inject a response writer that tracks all necessary information written to the response, to be able to persist this information. That way we can return the same responses multiple times without doing any more work.